client/src/components/calendar.jsx

```diff
@@ -1,20 +1,64 @@
 import React from 'react';
+import moment from 'moment';
+import Month from './month.jsx';
 
 class Calendar extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            current: null
+            current: moment().month(),
+            first: moment.months()[moment().month()],
+            second: moment.months()[(moment().month()) + 1]
         }
+        this.advanceMonth = this.advanceMonth.bind(this);
+        this.previousMonth = this.previousMonth.bind(this);
+    }
+
+    renderFirstDay (month) {
+        return moment().month(month).date(1).format("d");
+    }
+
+    getDaysInMonth (month) {
+        return moment().month(month).daysInMonth();
+    }
+
+    advanceMonth () {
+        const newMonth = this.state.current + 1;
+        this.setState({
+            current: newMonth,
+            first: moment.months()[newMonth],
+            second: moment.months()[newMonth + 1]
```

Comment:
This function increments the currentMonth counter, automatically triggering the re-render with the correct month (old + 1)

Reply:
Nice!